### PR TITLE
cp-268: Delete chat functionality

### PR DESCRIPTION
**Pull request by OlenaYatseniuk**
- Add delete methods in chat controller, service and repository
- Add delete logic in frontend (delete actions and slices, delete icon, deleteChatModal) 

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/102380564/3f693ccd-abb5-44b5-bcbc-1e8c5d72efc3

### CODE REVIEW

**Comment by liza-veis** [comment]
Can we just make icon color deeper instead of adding background on hover?

File: backend/src/packages/chats/chat.controller.ts [hunk 1]
```diff
@@ -153,6 +175,45 @@ class ChatController extends BaseController {
...
+   *      - bearerAuth: []
+   *     responses:
+   *       200:
+   *         description: Successful operation
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: boolean
+   *       404:
+   *         description: User was not found
```
Comment (OlesiaTr) [inline backend/src/packages/chats/chat.controller.ts]:
```suggestion
   *         description: Chat was not found
```

File: frontend/src/libs/components/icon/icon.tsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
-import { type IconName } from '#libs/types/types.js';
+import { type IconColor } from '#libs/enums/icon-color.enum.js';
```
Comment (OlesiaTr) [inline frontend/src/libs/components/icon/icon.tsx]:
```suggestion
import { type IconColor } from '#libs/enums/enums.js';
```

**Comment by OlenaYatseniuk** [comment]
> Can we just make icon color deeper instead of adding background on hover?

Done

**Comment by OlenaYatseniuk** [comment]
Had too much merge conflict that I didn't manage to resolve, so I created a new one pr #327

**Review by OlesiaTr — APPROVED** [review]

**Pull request closed without merging**